### PR TITLE
Implementation of more commands.

### PR DESCRIPTION
**Pull request by WeomuCat**
Implements these features:
#22 #21

### CODE REVIEW

**Comment by bruceskellator** [comment]
Denis, for my implementation of undo, it requires state-changing commands (e.g. Add, Delete, Clear) to call a clearFutureStates method (that clears UndoableHistory of all EventList after the pointer) because after we execute a normal command, all future states are no longer needed.  
After that, they call a commit method which duplicates the mainEventList and stores the copied version to the history.
Just informing you on my approach since it affects the command classes. I can do the changes that are needed.

**Comment by WeomuCat** [comment]
> Denis, for my implementation of undo, it requires state-changing commands (e.g. Add, Delete, Clear) to call a clearFutureStates method (that clears UndoableHistory of all EventList after the pointer) because after we execute a normal command, all future states are no longer needed.
> After that, they call a commit method which duplicates the mainEventList and stores the copied version to the history.
> Just informing you on my approach since it affects the command classes. I can do the changes that are needed.

Ah I see, this should be handled in ModelManager.
Modify ModelManager's state changing methods (`.addEvent()`, `.deleteEvent()`) to include `clearFutureHistory()` before `commitToHistory()`.
Or even better, why not add `clearFutureHistory()` directly in `commit()`?

File: src/main/java/seedu/address/logic/commands/CommandBuilder.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    private Option context;
+
+    // Represents the command arguments in a Command.
+    private Option arguments;
+    private Map<String, Option> options;
+
+    /**
+     * This method must be called first.
+     */
+    CommandBuilder init() {
```
Comment (marcusteh1238) [inline src/main/java/seedu/address/logic/commands/CommandBuilder.java]: Missing access modifier

**Comment by bruceskellator** [comment]
> > Denis, for my implementation of undo, it requires state-changing commands (e.g. Add, Delete, Clear) to call a clearFutureStates method (that clears UndoableHistory of all EventList after the pointer) because after we execute a normal command, all future states are no longer needed.
> > After that, they call a commit method which duplicates the mainEventList and stores the copied version to the history.
> > Just informing you on my approach since it affects the command classes. I can do the changes that are needed.
> 
> Ah I see, this should be handled in ModelManager.
> Modify ModelManager's state changing methods (`.addEvent()`, `.deleteEvent()`) to include `clearFutureHistory()` before `commitToHistory()`.
> Or even better, why not add `clearFutureHistory()` directly in `commit()`?

@WeomuCat Okay noted.

**Pull request merged**